File: C1.UWP.Maps/CS/MapsSamples/Samples/VectorLayer/ShapeFileHelper.cs
using C1.Xaml.Maps;
using System.Collections.Generic;
using System.IO;

namespace MapsSamples
{
    public delegate void ProcessShapeItem(C1VectorItemBase vector, C1ShapeAttributes attributes, Location location);

    public static class ShapeFileHelper
    {
        public static void LoadShape(this C1VectorLayer vl, Stream stream, Stream dbfStream, Location location, bool centerAndZoom,
      ProcessShapeItem processShape)
        {
            Dictionary<C1VectorItemBase, C1ShapeAttributes> vects = ShapeReader.Read(stream, dbfStream);

            vl.BeginUpdate();
            foreach (C1VectorItemBase vect in vects.Keys)
            {
                if (processShape != null)
                {
                    processShape(vect, vects[vect], location);
                }

                vl.Children.Add(vect);
            }
            vects.Clear();
            vl.EndUpdate();
        }
    }
}

File: C1.UWP.Maps/VB/MapsSamples/Samples/VectorLayer/ShapeFileHelper.vb
Imports System.IO
Imports System.Collections.Generic
Imports C1.Xaml.Maps

Public Delegate Sub ProcessShapeItem(vector As C1VectorItemBase, attributes As C1ShapeAttributes, location As Location)

Public Module ShapeFileHelper
    <Extension()>
    Public Sub LoadShape(vl As C1VectorLayer, stream As Stream, dbfStream As Stream, location As Location, centerAndZoom As Boolean, processShape As ProcessShapeItem)
        Dim vects As Dictionary(Of C1VectorItemBase, C1ShapeAttributes) = ShapeReader.Read(stream, dbfStream)

        vl.BeginUpdate()
        For Each vect As C1VectorItemBase In vects.Keys
            If processShape IsNot Nothing Then
                processShape(vect, vects(vect), location)
            End If

            vl.Children.Add(vect)
        Next
        vects.Clear()
        vl.EndUpdate()
    End Sub
End Module